iroh/src/util/fs.rs

```diff
@@ -225,10 +226,48 @@ fn path_content_info0(path: impl AsRef<Path>) -> anyhow::Result<PathContent> {
     Ok(PathContent { size, files })
 }
 
+/// Helper function that translates a path on a filesystem to a key in a document.
+pub fn path_to_key(path: impl AsRef<Path>) -> anyhow::Result<Bytes> {
+    let buf = path
+        .as_ref()
+        .to_str()
+        .map(|p| p.as_bytes())
+        .ok_or(anyhow::anyhow!("could not convert path to bytes"))?;
+    let mut bytes = BytesMut::from(buf);
+    bytes.put(&b"\0"[..]);
+    Ok(bytes.freeze())
+}
+
+/// Helper function that translates a key that was derived from a path on a filesystem back to that
+/// path.
+pub fn key_to_path(key: impl AsRef<[u8]>) -> anyhow::Result<PathBuf> {
+    let key = std::str::from_utf8(key.as_ref())?;
```

Comment:
we shouldn't assume utf8 here, this should use OsString, and the nul byte should be removed before conversion, to avoid accidental issues